mentat/code_file_manager.py

```diff
@@ -86,14 +101,20 @@ async def write_changes_to_files(
                 )
                 if await ask_yes_no(default_yes=False):
                     await stream.send(f"Deleting {rel_path}...", color="red")
+                    # We use the current lines rather than the stored lines for undo
```

Comment:
What's the difference between "current lines" and "stored lines" here?

Reply:
We store the lines when we send the user's input to mentat; we do this because mentat's changes won't work correctly if the user changes the file (you can see in this same function we also check if the file was changed; if it was, we ask the user if they're sure they want their changes erased). But if you apply mentat's changes over your changes, I think it makes sense that you'd want undo to bring back your overwritten changes, rather than the stored lines. Let me know if you think we should do something else though, I'm open to suggestions!

Reply:
Ah gotcha, thanks for the clarification! Yeah I think that behavior makes sense too